### PR TITLE
Adding setup scripts for Linux and Windows

### PR DESCRIPTION
**Pull request by AlexOConnorHub**
It took me a second to figure out what all needed to be `pip3 install`ed. Thought I would make it easier for others :)

### CODE REVIEW

**Comment by nametable** [comment]
I appreciate your work, but basically what the scripts you have made do is the same as what I have on the [README.md](https://github.com/nametable/sau-daily-survey-bot/blob/master/README.md) when it says to run `pip install -r requirements.txt`. The file `requirements.txt` has the same 4 dependencies in it that your scripts have, so having both the scripts and `requirements.txt` would be redundant. However, `furl` is not actually needed, I just copied that import from a different project. So I can actually remove that dependency altogether.

**Comment by nametable** [comment]
However, a setup script might be nice if it asked the user for email/password and then made the `secrets.json` file for them. I guess it could also install dependencies at the same time... Hmm.

**Comment by AlexOConnorHub** [comment]
Oh, I guess I didn't read the README.md. That's why I made that script to install, because I'm a good programmer and do not read documentation.

**Comment by nametable** [comment]
Are you going to do anything more with your scripts PR @AlexOConnorHub or should I close? As of now, it would just be redundant to merge.

**Comment by AlexOConnorHub** [comment]
I'll work on adding making the .json file.

**Comment by nametable** [comment]
@AlexOConnorHub As a side note, since python is a prereq anyways, you could just make the setup script in python tooo... if you want. Then just add `#!/usr/bin/env python3` to the top of it.

File: setup/setup.sh [hunk 1]
```diff
@@ -1,2 +1,10 @@
 #!/bin/bash
+DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
 pip3 install requests furl bs4 lxml
```
Comment (nametable) [inline setup/setup.sh]: I recommend replacing this with `pip3 install -r requirements.txt` which will then mean that dependency changes will only need to reflected in one file, `requirements.txt`. This file is fairly standard for projects https://medium.com/@boscacci/why-and-how-to-make-a-requirements-txt-f329c685181e

File: setup/setup.sh [hunk 2]
```diff
@@ -1,2 +1,10 @@
...
+echo Please enter your SAU username: 
+read username
+echo Please enter your SAU passphrase: 
+read passphrase
+printf "{\n    \"username\":\"%s\",\n" $username > $DIR/../secrets.json
+printf "    \"password\":\"%s\"\n}" $passphrase >> $DIR/../secrets.json
```
Comment (nametable) [inline setup/setup.sh]: Does the `$DIR/../` end up being the directory in which the script is?

**Review by nametable — CHANGES_REQUESTED** [review]
Overall, looks good! Thanks Alex. Please switch to using `requirements.txt` though if you could, just to keep dependencies centralized.

File: setup/setup.sh (hunk 2 above)
Comment (AlexOConnorHub) [reply setup/setup.sh]: It does not. `$DIR/..` is one directory up. the following `/` is for making a `.json` file.

File: setup/setup.sh (hunk 1 above)
Comment (AlexOConnorHub) [reply setup/setup.sh]: Just switched both scripts to that.

**Review by nametable — APPROVED** [review]

**Comment by AlexOConnorHub** [comment]
Ok, the .sh works, and I tested the .bat on a win computer before changing it to use requirements.txt. I think it will work, but haven't texted.

File: setup/setup.sh [hunk 3]
```diff
@@ -1,10 +1,10 @@
 #!/bin/bash
-DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )/.."
+DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
```
Comment (nametable) [inline setup/setup.sh]: What was wrong with it?

Comment (AlexOConnorHub) [reply setup/setup.sh]: IDK, but it caused errors

Comment (nametable) [reply setup/setup.sh]: I think it was probably just line 3 maybe since you had added `/..` but then not removed it from the `pip3` command maybe...

Comment (nametable) [reply setup/setup.sh]: Or was there more errors too?

Comment (AlexOConnorHub) [reply setup/setup.sh]: Wait... I forgot to change the pip3 line. It works now!

**Comment by nametable** [comment]
I guess that's probably good then unless you want to update `README.md`, but I can do that after merge if you want.

**Comment by AlexOConnorHub** [comment]
I think that's good for the README.md. You agree?

**Comment by nametable** [comment]
> I think that's good for the README.md. You agree?

Seems fine. On Windows do they have to `cd` into the folder for it to work?

**Comment by AlexOConnorHub** [comment]
They can just click on it in File Explorer.

**Comment by nametable** [comment]
> They can just click on it in File Explorer.

Oh, right. OK I'll merge now. Good job!

**Comment by AlexOConnorHub** [comment]
Thanks!

**Pull request merged**